src/azul/plugins/repository/tdr/__init__.py

```diff
@@ -292,14 +317,14 @@ def _retrieve_entities(self,
         return rows
 
     def _parse_links(self,
```

Comment:
I think, this really wants to be a method of the Links class. The current state on this branch is also acceptable so we should add another, separate commit for this refactoring. Not sure what's going to happen to the `for_project` factory. It might be that method or it might just dissolve.